src/chronicle_storage.erl

```diff
@@ -158,25 +170,26 @@ open(PublishSnapshotFun) ->
 
     publish(Storage5).
 
-open_logs(#storage{data_dir = DataDir} = Storage) ->
-    {Orphans, Sealed, Current} = find_logs(DataDir),
-
-    NewStorage0 = Storage#storage{meta = #{},
-                                  pending_meta = #{},
-                                  config = undefined,
-                                  committed_config = undefined,
-                                  pending_configs = [],
-                                  low_seqno = ?NO_SEQNO + 1,
-                                  high_seqno = ?NO_SEQNO,
-                                  pending_high_seqno = ?NO_SEQNO,
-                                  current_snapshot = no_snapshot,
-                                  extra_snapshots = [],
-                                  log_segments = []},
-
-    %% TODO: Be more robust when dealing with corrupt log files. As long as a
-    %% consistent state can be recovered, corrupt logs should not be a fatal
-    %% error.
-    NewStorage1 =
+reset_storage(Storage) ->
+    ets:delete_all_objects(Storage#storage.log_tab),
+    ets:delete_all_objects(Storage#storage.log_info_tab),
+    Storage#storage{meta = #{},
+                    pending_meta = #{},
+                    config = undefined,
+                    committed_config = undefined,
+                    pending_configs = [],
+                    low_seqno = ?NO_SEQNO + 1,
+                    high_seqno = ?NO_SEQNO,
+                    pending_high_seqno = ?NO_SEQNO,
+                    current_snapshot = no_snapshot,
+                    extra_snapshots = [],
+                    log_segments = []}.
+
+open_logs(Sealed, Current, Storage) ->
```

Comment:
Can I request a function comment on open_logs? Something like the following:

> Opens, reads and validates the supplied chronicle log files exiting if there is a problem. Upon successful return of this function (or a trapped exit) the log_tab and log_info_tab ets tables will be populated with relevant information.

Reply:
Done